integration/spark/README.md

```diff
@@ -72,61 +72,72 @@ spark = (SparkSession.builder.master('local').appName('rdd_to_dataframe')
+
+##### URL
 
-| Parameter | Definition | Example |
-------------|------------|---------
-| spark.openlineage.transport.kinesis.streamName | Required, the streamName of the Kinesis Stream | some-stream-name |
-| spark.openlineage.transport.kinesis.region | Required, the region of the stream | us-east-2 |
-| spark.openlineage.transport.kinesis.roleArn | Optional, the roleArn which is allowed to read/write to Kinesis stream | some-role-arn |
-| spark.openlineage.transport.kinesis.[xxx] | Optional, the [xxx] is property of [Kinesis allowd properties](https://github.com/awslabs/amazon-kinesis-producer/blob/master/java/amazon-kinesis-producer-sample/default_config.properties) | 1 |
+You can supply http parameters using values in url, the parsed `spark.openlineage.*` properties are located in url as follows:
 
-#### URL
+`{transport.url}/{transport.endpoint}/namespaces/{namespace}/jobs/{parentJobName}/runs/{parentRunId}?app_name={appName}&api_key={transport.apiKey}&timeout={transport.timeout}&xxx={transport.urlParams.xxx}`
+
+example:
+
+`http://localhost:5000/api/v1/namespaces/ns_name/jobs/job_name/runs/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx?app_name=app&api_key=abc&timeout=5000&xxx=xxx`
+
+### Kinesis 
+If `spark.openlineage.transport.type` is set to `kinesis`, then the below parameters would be read and used when building KinesisProducer.
+Also, KinesisTransport depends on you to provide artifact `com.amazonaws:amazon-kinesis-producer:0.14.0` or compatible on your classpath.
 
-`spark.openlineage.url` covers `spark.openlineage.*` parameters in the form of resources and URL parameters.
+| Parameter                                     | Definition                                                                                                                                                                                   | Example          |
+-----------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|------------------
+| spark.openlineage.transport.streamName        | Required, the streamName of the Kinesis Stream                                                                                                                                               | some-stream-name |
+| spark.openlineage.transport.region            | Required, the region of the stream                                                                                                                                                           | us-east-2        |
+| spark.openlineage.transport.roleArn           | Optional, the roleArn which is allowed to read/write to Kinesis stream                                                                                                                       | some-role-arn    |
```

Comment:
is this still optional?

Reply:
it's no longer type `Optional` but yeah, it's not a required property